### PR TITLE
UI services updates

### PR DESCRIPTION
**Pull request by sbenedicadb**
- animations, gestures, takeover code in MessageSettings and FullscreenMessage

### CODE REVIEW

File: AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift [hunk 1]
```diff
@@ -191,6 +193,11 @@ public class FullscreenMessage: NSObject, FullscreenPresentable {
     ///   - name: the name of the message being passed from javascript
     ///   - handler: a method to be called when the javascript message is passed
     public func handleJavascriptMessage(_ name: String, withHandler handler: @escaping (Any?) -> Void) {
+        // don't add the handler if it's already been added
```
Comment (praveek) [inline AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift:196]: Shouldn't this be executed in DispatchQueue.main ?

File: AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift [hunk 2]
```diff
@@ -227,153 +234,109 @@ public class FullscreenMessage: NSObject, FullscreenPresentable {
...
     }
 
-    // width in settings represents a percentage of the screen.
-    // e.g. - 80 = 80% of the screen width
-    // default value is full screen width
-    private var width: CGFloat {
-        if let settingsWidth = settings?.width {
-            return screenWidth * CGFloat(settingsWidth) / 100
+    @objc func handleGesture(_ sender: UIGestureRecognizer? = nil) {
+        guard let recognizer = sender as? MessageGestureRecognizer else {
```
Comment (praveek) [inline AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift:265]: Shouldn't this be executed in the context of DispatchQueue.main?

File: AEPCore/Tests/EventHubTests/EventTests.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/*
+ Copyright 2020 Adobe. All rights reserved.
```
Comment (rymorale) [inline AEPCore/Tests/EventHubTests/EventTests.swift:2]: should be 2021

File: AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+ */
+
+import Foundation
+import UIKit
+
+extension FullscreenMessage {
+    // MARK: - internal vars
+
+    /// calculated frames are used for animation
+    /// frameWhenVisible is the frame when the message is visible to the user
```
Comment (rymorale) [inline AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift:20]: would it be clearer if the "frameWhenVisible..." comment came before the "calculated frames..." comment?

File: AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        case .left:
+            return CGRect(x: -(screenWidth + originX), y: originY, width: width, height: height)
+        case .center:
+            return CGRect(x: screenWidth * 0.5, y: screenHeight * 0.5, width: 0, height: 0)
+        default:
+            return frameWhenVisible
+        }
+    }
+
+    // returns the width of the screen, measured in points
```
Comment (rymorale) [inline AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift:69]: missing `/` in the doc comment

File: AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+    var screenHeight: CGFloat {
+        return UIScreen.main.bounds.height
+    }
+
+    // MARK: - private vars
+
+    // width in settings represents a percentage of the screen.
+    // e.g. - 80 = 80% of the screen width
+    // default value is full screen width
```
Comment (rymorale) [inline AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift:82]: should these be doc comments?

Comment (rymorale) [reply AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift:82]: additionally, for ease of reading i think periods can be added to the end of the sentences which are missing them.

**Review by praveek — APPROVED** [review]

**Pull request merged**